### DISCUSSION TITLE
2.6 Bug with ValidatesInput trait

### DISCUSSION THREAD

**Discussion by delta1186** [livewire/livewire#3720] [7. Bugs (archived)] [answered]
@calebporzio I just updated to 2.6 and I think you have a bad Illuminate method reference. Here is the error I am getting.  
![image](https://user-images.githubusercontent.com/421307/133313024-31e919ec-2e1f-4fe1-af55-48775e109713.png)

I checked the laravel api and it is different.  I am running the latest release of laravel too.  I searched for getBag and this was the only reference I got back.
https://laravel.com/api/8.x/Illuminate/Support/ViewErrorBag.html#method_getBag 

I believe this is from `use ComponentConcerns\ValidatesInput` in src/Component.php.  If you look in that trait it is using `use Illuminate\Support\MessageBag;` and that class does not have `getBag` instead it is `getMessageBag`  I manually changed `getBag` to `getMessageBag` and it fixed the issue.

**Comment by joshhanley** [comment]
@delta1186 thanks for reporting! Can you provide steps that you followed to replicate the issue, and any relevant code please? 

Thanks!

**Reply by delta1186** [reply]
Okay here we go.  I hope this is enough, but just let me know. :)

I am using the spatie media library pro and have a livewire component with one media library attachment component.  
```
<div class="mt-5">
    <x-input.group inline="true" label="Add attachments" for="attachments" :error="$errors->first('attachments')">
        <x-media-library-attachment name="attachments" rules="mimes:png,jpeg,pdf,doc,docx,xls,xlsx,gif" multiple />
    </x-input.group>
</div>
```
In my livewire component I use a trait `WithMedia`.  That trait has the following method.
```
public function renderingWithMedia(): void
    {
        $errorBag = $this->getErrorBag();

        foreach ($this->getMediaComponentNames() as $mediaComponentName) {
            $this->emit('mediaComponentValidationErrors', $mediaComponentName, $errorBag->toArray());
        }
    }
```
which then fires this
```
public function onMediaComponentValidationErrors(string $componentName, array $validationErrors)
    {
        if ($componentName !== $this->name) {
            return;
        }

        $messageBag = new MessageBag($validationErrors);

        if ($messageBag->has($this->name)) {
            $this->onShowListErrorMessage($messageBag->first($this->name));
        } else {
            $this->listErrorMessage = null;
        }

        $this->validationErrors = $messageBag;
    }
```
But all this is doing is setting the error bag.  Which now throws the exception after the following commit.  I reverted back this line and it worked.

![image](https://user-images.githubusercontent.com/421307/133345492-6acd603b-4e31-4d9b-af23-f0015a0cb70b.png)

https://github.com/livewire/livewire/commit/4720f5abe8a9d1f89cec752497acbff09422a2d2#diff-79446737c5fc656d037e5ebaee2e336acb25c73dfff3d03856d42728276b9b32R170

So all this to say this commit should have used `getMessageBag `  because `getBag` is not a method on `Illuminate\Support\MessageBag;`

**Comment by joshhanley** [comment, marked as answer]
@delta1186 I have tagged v2.6.2 which fixes this issue 🙂

Please test it and let us know if it's still happening!

Thanks!

**Reply by delta1186** [reply]
@joshhanley yeah that worked.  Hate that you had to revert it back to the original line, but probably makes sense.  I think the author thought they were using an instance of `ViewErrorBag` since it is referenced at the top of the component class, so in that case `getBag('default')` would have been right.  However when reviewing what `getErrorBag` returns you see it is from the ValidatesInput concern and it returns a `MessageBag` and to get the default message bag you would have just called `getMessageBad()` which does take any arguments.  Maybe someone can resubmit that as a new PR since it appears they were trying to fix a bug, but I guess they tests the added did not cover that change.

#3289  Fix bug where view get data errors doesn't return an `ErrorBag`

Haha, I actually just realized that was actually your commit.  Whoops. :)

**Reply by joshhanley** [reply]
@delta1186 haha yep that was me, hence why I knew I could safely revert it.

The issue was, an error was being thrown in one of Livewire's underlying view, and that was returning an instance of `ViewErrorBag` hence why I needed to call `getErrorBag` as `getMessageBag` isn't on that object.

But it caused a breaking change when what was received was already a `MessageBag` like you were experiencing. And it was no longer needed as we refactored the feature that was using it, so I could just simply revert it without any issues 🙂
